### PR TITLE
Create class Person, Teacher, Student

### PR DESCRIPTION
**Pull request by bienvenuushindi**
- [ ] Create class Person (states and methods)
- [ ] Create class Teacher (states and methods)
- [ ] Create class Student (states and methods)

### CODE REVIEW

**Review by TedLivist — COMMENTED** [review]
## STATUS: INVALID CODE REVIEW REQUEST 🛑 🛑

Hi @bienvenuushindi,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

### You do not have a descriptive README. Kindly format your README using [the Microverse template](https://github.com/microverseinc/readme-template)

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: classes/student.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class Student < Person
+  def initialize(classroom, age, name)
+    super(age, name)
+    @classroom = classroom
+  end
```
Comment (omar-muhamad) [inline classes/student.rb]: - [ ] As per this [requirement](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_people.md#:~:text=Constructor%20with%20name%2C%20age%2C%20and%20parent_permission%20as%20parameter.%20name%20and%20parent_permission%20are%20optional%20and%20have%20default%20values%20of%20%22Unknown%22%20and%20true.), Please add `parent_permission` as parameter and ensure that `name` and `parent_permission` have default values of "Unknown" and true.

File: classes/teacher.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class Teacher < Person
+  def initialize(specialization, age, name)
+    super(age, name)
+    @specialization = specialization
+  end
```
Comment (omar-muhamad) [inline classes/teacher.rb]: - [ ] Same Here, As per this [requirement](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_people.md#:~:text=Constructor%20with%20name%2C%20age%2C%20and%20parent_permission%20as%20parameter.%20name%20and%20parent_permission%20are%20optional%20and%20have%20default%20values%20of%20%22Unknown%22%20and%20true.), Please add `parent_permission` as parameter and ensure that `name` and `parent_permission` have default values of "Unknown" and true.

File: classes/person.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
+class Person
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  attr_reader :id
+
+  attr_accessor :name, :age
```
Comment (omar-muhamad) [inline classes/person.rb]: - [ ] [Optional] It's considered a best practice if you add these setters and getters values before the `initialize` method.

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @bienvenuushindi,

Nice job, you are almost there.



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- Gitflow is used correctly. ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below.

## [Optional] suggestions:

- [ ] Please check the inline comments below.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: classes/person.rb (hunk 3 above)
Comment (sja-thedude) [reply classes/person.rb]: Implemented 👍

Comment (bienvenuushindi) [reply classes/person.rb]: @sja-thedude Thanks :bow:

File: classes/student.rb [hunk 4]
```diff
@@ -1,6 +1,6 @@
 class Student < Person
-  def initialize(classroom, age, name)
-    super(age, name)
+  def initialize(classroom, age, name, parent_permission)
+    super(age, name, parent_permission: parent_permission)
```
Comment (sja-thedude) [inline classes/student.rb]: - [ ] Kindly make sure to implement the changes asked by the previous reviewer. Follow below:
```ruby
def initialize(age, classroom, name = "Unknown", parent_permission = true)  
    super(age, name, parent_permission)
    @classroom = classroom
end
```

File: classes/teacher.rb [hunk 5]
```diff
@@ -1,6 +1,6 @@
 class Teacher < Person
-  def initialize(specialization, age, name)
-    super(age, name)
+  def initialize(specialization, age, name, parent_permission)
+    super(age, name, parent_permission: parent_permission)
```
Comment (sja-thedude) [inline classes/teacher.rb]: - [ ] Kindly make sure to implement the changes asked by the previous reviewer. Follow below:
```ruby
def initialize(age, specialization, name = "Unknown", parent_permission = true)
    super(age, name, parent_permission)
    @specialization = specialization
  end
```

**Review by sja-thedude — CHANGES_REQUESTED** [review]
## Required Changes ♻️

Hi @bienvenuushindi, This is @sja-thedude your code reviewer.

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

200w

 ## Highlights
✔️ Good job so far

 ## Required Changes ♻️
- [ ] _Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: classes/teacher.rb (hunk 5 above)
Comment (bienvenuushindi) [reply classes/teacher.rb]: @sja-thedude Thank you 
But my rubocop linter complains when I assign a default value of type boolean to `parent_permission`  with the `=` sign. 
It says that I should use instead `parameter: default_value`.

**Review by codepantha — APPROVED** [review]
# STATUS: APPROVED 🟢 ✔️ 

## Hi @bienvenuushindi 👋, 

Well done for implementing the required changes 👏 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**